libraries/program-error/derive/src/macro_impl.rs

```diff
@@ -114,3 +144,53 @@ pub fn spl_program_error(input: ItemEnum) -> proc_macro2::TokenStream {
         #print_program_error
     }
 }
+
+/// This function adds a discriminant to the first enum variant based on the
+/// hash of the `SPL_ERROR_HASH_NAMESPACE` constant, the enum name and variant
+/// name.
+/// It will then check to make sure the provided `hash_error_code_start` is
+/// equal to the hash-produced `u32`.
+///
+/// See https://docs.rs/syn/latest/syn/struct.Variant.html
+fn set_first_discriminant(item_enum: &mut ItemEnum, error_code_start: u32) {
+    let enum_ident = &item_enum.ident;
+    let first_variant = &mut item_enum.variants[0];
+    let discriminant = u32_from_hash(enum_ident, &first_variant.ident);
+    if discriminant == error_code_start {
+        let eq = Token![=](Span::call_site());
+        let expr = Expr::Lit(ExprLit {
+            attrs: Vec::new(),
+            lit: Lit::Int(LitInt::new(&discriminant.to_string(), Span::call_site())),
+        });
+        first_variant.discriminant = Some((eq, expr));
+    } else {
+        panic!(
+            "Error code start value from hash is {}. Update your macro attribute.",
```

Comment:
Let's tell people *exactly* what to do
```suggestion
            "Error code start value from hash must be {0}. Update your macro use to `#[spl_program_error(hash_error_code_start = {0})]`.",
```